modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb

```diff
@@ -175,7 +175,7 @@ def run_host(ip)
           create_credential_login(credential_data)
           tmp_key = result.credential.private
           ssh_key = SSHKey.new tmp_key
-          session_setup(result, scanner, ssh_key.fingerprint, credential_core.private_id) if datastore['CreateSession']
+          session_setup(result, scanner, ssh_key.fingerprint, result.credential.private) if datastore['CreateSession']
```

Comment:
Hey so doing some further digging and it looks like the current code is correct _if_ you have an active database connection, but of course you aren't guaranteed to have a database connection so we'd probably need something like this to guard against this problem
```suggestion
          if datastore['CreateSession']
            if credential_core.is_a? Metasploit::Credential::Core
              session_setup(result, scanner, ssh_key.fingerprint, credential_core.private_id)
            else
              session_setup(result, scanner, ssh_key.fingerprint, nil)
            end
          end
```

Reply:
I tested the suggested change using my previous test case (without a database), and it worked - no error and the session was correctly created using the valid username and key.

Reply:
@stephenbradshaw fantastic! would you like to update your pr to reflect the changes?